### PR TITLE
Correctly extend a lognormed colorbar

### LINKED ISSUE #5201: issue with colorbar using LogNorm and extend='min'
I often create log-scaled pcolormesh plots on my (university) computer and frequently encounter an issue: whenever I plot a colorbar using `extend='both'` or `extend='min'` and a LogNorm with vmin < 1.0, the bottom triangle of the colorbar stays white, no matter what colormap I use.

Here is a piece of code that recreates the problem and also shows an example where the colorbar looks like it should (the difference is that vmin >= 1.0).

``` python
import matplotlib as mpl
import matplotlib.pyplot as plt

norm0 = mpl.colors.LogNorm(vmin=1.0, vmax=1000.0)
norm1 = mpl.colors.LogNorm(vmin=0.1, vmax=1000.0)

fig, axs = plt.subplots(1,2)

# no problem here
mpl.colorbar.ColorbarBase(axs[0], norm=norm0, orientation='vertical', extend='both')
# bottom triangle section of color bar stays white
mpl.colorbar.ColorbarBase(axs[1], norm=norm1, orientation='vertical', extend='both')

plt.show()
```

matplotlib_logcolorbar

I am using matplotlib version 1.2.1 (and I have also recreated the problem in 1.3.1). If this apparent bug has been fixed in later versions, is there an easy workaround to fill the triangle?  Unfortunately I cannot easily update the software on university computers.

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #5201.

### CODE REVIEW

**Comment by tacaswell** [comment]
Those failures look real?

Is there a better way to do this that does not require `isinstance`?

**Comment by dstansby** [comment]
isinstace is used quite a bit nearby so I assumed it would be fine.

Turns out the issue is a typo (+ should be a -), in the non-lognorm code I copied into the else statement. Should be fixed now.

**Comment by efiring** [comment]
`isinstance` is fine here.  This looks good to me.  I think it should be merged and backported.  The appveyor tests pass and Travis is simply stalled.

File: lib/matplotlib/tests/test_colorbar.py [hunk 1]
```diff
@@ -325,6 +325,14 @@ def test_colorbar_get_ticks():
...
 
 
+@cleanup
+def test_colorbar_lognorm_extension():
+    # Test that colorbar with lognorm is extended correctly
+    ax = plt.gca()
+    cb = ColorbarBase(ax, norm=LogNorm(vmin=0.1, vmax=1000.0),
+                      orientation='vertical', extend='both')
+    assert cb._values[0] >= 0.0
+
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_colorbar.py]: One more blank line.

File: lib/matplotlib/tests/test_colorbar.py [hunk 2]
```diff
@@ -325,6 +325,14 @@ def test_colorbar_get_ticks():
     assert defTicks.get_ticks().tolist() == levels
 
 
+@cleanup
+def test_colorbar_lognorm_extension():
+    # Test that colorbar with lognorm is extended correctly
+    ax = plt.gca()
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_colorbar.py]: Better to use `plt.subplots`.

**Comment by tacaswell** [comment]
restarted travis just to be sure.

**Comment by efiring** [comment]
Appveyor failures look unrelated.

**Comment by efiring** [comment]
Backported to v2.x as 58b92b2.

**Pull request merged**